lib/smart_proxy_dynflow_core/log.rb

```diff
@@ -40,6 +40,25 @@ def log_file
       end
     end
 
+    def initialize(file, *rest)
+      @fd = file.kind_of?(IO) ? file : File.open(file, 'a')
+      super(@fd, rest)
+    end
+
+    def add(*args)
+      handle_log_rolling if @roll_log
+      super(*args)
+    end
+
+    def roll_log
+      @roll_log = true
+    end
+
+    def handle_log_rolling
+      @roll_log = false
+      @fd.reopen unless log_file.kind_of? IO
```

Comment:
This is defined on class, while we are on instance: this doesn't work. Why not just saving the @file in initializer in case it's string and using it as condition + reopen needs file and mode specified, otherwise it doesn't work.